x-pack/platform/plugins/private/cross_cluster_replication/server/routes/api/follower_index/register_create_route.ts

```diff
@@ -52,7 +52,9 @@ export const registerCreateRoute = ({
     license.guardApiRoute(async (context, request, response) => {
       const { client } = (await context.core).elasticsearch;
       const { name, ...rest } = request.body;
-      const body = removeEmptyFields(serializeFollowerIndex(rest as FollowerIndex));
+      const body = removeEmptyFields(
+        serializeFollowerIndex(rest as FollowerIndex)
+      ) as FollowerIndexToEs;
```

Comment:
`removeEmptyFields` returns `Record<string, any>` which is not compliant with `FollowerIndexToEs`